File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();

        int mn = prices[0];
        int mx = 0;

        for(int i = 1 ; i < n ; i++){
            mx = max(mx, prices[i]-mn);
            mn = min(mn, prices[i]);
        }

        return mx;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, nums: List[int]) -> int:
        n = len(nums)
        l, r = 0, n-1

        mx = -1

        while l < r:
            mx = max(mx, min(nums[l], nums[r]) * (r-l))

            if nums[l] < nums[r]:
                l += 1
            
            elif nums[r] < nums[l]:
                r -= 1
            
            else: 
                l += 1
                r -= 1
        

        return mx
